### ISSUE TITLE
Document alternative approach to testing by creating arbitrary req/resp/ws

### ISSUE THREAD

**Issue by vytas7** [falconry/falcon#1793]
New methods were introduced as part of https://github.com/falconry/falcon/issues/1506 that afford an alternative approach to testing by creating arbitrary req/resp/ws:
* [`create_req`](https://falcon.readthedocs.io/en/latest/api/testing.html#falcon.testing.create_req)
* [`create_asgi_req`](https://falcon.readthedocs.io/en/latest/api/testing.html#falcon.testing.create_asgi_req)
* [`create_scope_ws`](https://falcon.readthedocs.io/en/latest/api/testing.html#falcon.testing.create_scope_ws)

Write a paragraph on how to use these goodies.
From https://github.com/falconry/falcon/issues/1506:
```python
def test_my_resource():
    resource = MyResource()
    req, resp = Request(helpers.create_environ()), Response()
    resource.on_get(req, resp)
```

**Comment by CaselIT** [comment]
> Provide a global method in testing.helpers which creates a Request and Response object. 

I think it could also be useful if we also set the context with this function. Maybe by passing a dict that gets translated in the context object.

**Comment by vytas7** [comment]
@CaselIT maybe it would be enough to document it?
One can operate on the created `req` or `resp` as usual, i.e.:
```python
>>> req = falcon.testing.create_asgi_req(path='/things', method='PATCH')
>>> req.context.key = 'value'
>>> req.context
Context({'key': 'value'})
```

**Comment by CaselIT** [comment]
sure that works too, but it shouldn't be too hard to allow context to be passed as argument (unless `**kw` is already sent to another mapping where context may be a key used by people)

**Comment by vytas7** [comment]
I see, that might make sense. I don't think we should pipe `**kw` there though, because that might be ambiguous with setting env/scope key-values. But we can add another mapping argument.

**Comment by CaselIT** [comment]
no, I just meant that it should be fine to add a `context=None` there, unless we take `**kw` and use it to set another mapping. If that's the case then this would be a backward incompatible change since people could be using context to pass data somewhere else.

If that's the case then best course of action is to either not have it or use another keyword that's more unlikely, like `requext_context` or something

**Comment by vytas7** [comment]
Ah, no, these functions are incomplete (cannot set `body`, `json`, etc) and lack examples, it is absolutely not a breaking change to add things there.

**Comment by iuliachiriac** [comment]
I started looking into this (during EuroPython sprints) 😄
